SchemaParseException Thrown for ThincItSymbolCheckQuarterSummary records

the connector throws the following exception - 
```
[2018-06-13 16:49:33,943] INFO Opening record writer for: hdfs://hdfs-namenode:8020/topicAndroidNew//+tmp/thincit_symbol_check/partition=1/3ca45018-7b5a-4886-9127-b7e4a891020e_tmp.avro (org.radarcns.sink.hdfs.AvroRecordWriterProviderRadar)
[2018-06-13 16:49:33,962] ERROR WorkerSinkTask{id=radar-hdfs-sink-android-15000-3} Task threw an uncaught and unrecoverable exception. Task is being killed and will not recover until manually restarted. (org.apache.kafka.connect.runtime.WorkerSinkTask)
org.apache.avro.SchemaParseException: Can't redefine: org.radarcns.active.thincit.ThincItSymbolCheckQuarterSummary
	at org.apache.avro.Schema$Names.put(Schema.java:1128)
	at org.apache.avro.Schema$NamedSchema.writeNameRef(Schema.java:562)
	at org.apache.avro.Schema$RecordSchema.toJson(Schema.java:690)
	at org.apache.avro.Schema$RecordSchema.fieldsToJson(Schema.java:716)
	at org.apache.avro.Schema$RecordSchema.toJson(Schema.java:701)
	at org.apache.avro.Schema$RecordSchema.fieldsToJson(Schema.java:716)
	at org.apache.avro.Schema$RecordSchema.toJson(Schema.java:701)
	at org.apache.avro.Schema.toString(Schema.java:324)
	at org.apache.avro.Schema.toString(Schema.java:314)
	at org.apache.avro.file.DataFileWriter.create(DataFileWriter.java:144)
	at org.apache.avro.file.DataFileWriter.create(DataFileWriter.java:135)
	at org.radarcns.sink.hdfs.AvroRecordWriterProviderRadar$1.write(AvroRecordWriterProviderRadar.java:105)
	at io.confluent.connect.hdfs.TopicPartitionWriter.writeRecord(TopicPartitionWriter.java:643)
	at io.confluent.connect.hdfs.TopicPartitionWriter.write(TopicPartitionWriter.java:379)
	at io.confluent.connect.hdfs.DataWriter.write(DataWriter.java:374)
	at io.confluent.connect.hdfs.HdfsSinkTask.put(HdfsSinkTask.java:109)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:524)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:302)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:205)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:173)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:170)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:214)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2018-06-13 16:49:34,087] ERROR WorkerSinkTask{id=radar-hdfs-sink-android-15000-3} Task threw an uncaught and unrecoverable exception (org.apache.kafka.connect.runtime.WorkerTask)
org.apache.kafka.connect.errors.ConnectException: Exiting WorkerSinkTask due to unrecoverable exception.
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:546)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:302)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:205)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:173)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:170)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:214)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.apache.avro.SchemaParseException: Can't redefine: org.radarcns.active.thincit.ThincItSymbolCheckQuarterSummary
	at org.apache.avro.Schema$Names.put(Schema.java:1128)
	at org.apache.avro.Schema$NamedSchema.writeNameRef(Schema.java:562)
	at org.apache.avro.Schema$RecordSchema.toJson(Schema.java:690)
	at org.apache.avro.Schema$RecordSchema.fieldsToJson(Schema.java:716)
	at org.apache.avro.Schema$RecordSchema.toJson(Schema.java:701)
	at org.apache.avro.Schema$RecordSchema.fieldsToJson(Schema.java:716)
	at org.apache.avro.Schema$RecordSchema.toJson(Schema.java:701)
	at org.apache.avro.Schema.toString(Schema.java:324)
	at org.apache.avro.Schema.toString(Schema.java:314)
	at org.apache.avro.file.DataFileWriter.create(DataFileWriter.java:144)
	at org.apache.avro.file.DataFileWriter.create(DataFileWriter.java:135)
	at org.radarcns.sink.hdfs.AvroRecordWriterProviderRadar$1.write(AvroRecordWriterProviderRadar.java:105)
	at io.confluent.connect.hdfs.TopicPartitionWriter.writeRecord(TopicPartitionWriter.java:643)
	at io.confluent.connect.hdfs.TopicPartitionWriter.write(TopicPartitionWriter.java:379)
	at io.confluent.connect.hdfs.DataWriter.write(DataWriter.java:374)
	at io.confluent.connect.hdfs.HdfsSinkTask.put(HdfsSinkTask.java:109)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:524)
	... 10 more
```

This particularly happens because the parser is a singleton. There are 2 possible solutions - 
- Create a new parser each time (Or can even catch the above exception and then try creating a new parser)
- Put the definition of `ThincItSymbolCheckQuarterSummary` in a separate schema file. Right now its contained within `/commons/active/thincit/thinc_it_symbol_check.avsc`

--- a/src/main/java/org/radarcns/sink/hdfs/AvroFormatRadar.java
+++ b/src/main/java/org/radarcns/sink/hdfs/AvroFormatRadar.java
@@ -16,6 +16,8 @@
 
 package org.radarcns.sink.hdfs;
 
+import static io.confluent.connect.storage.StorageSinkConnectorConfig.SCHEMA_CACHE_SIZE_CONFIG;
+
 import io.confluent.connect.avro.AvroData;
 import io.confluent.connect.avro.AvroDataConfig;
 import io.confluent.connect.hdfs.HdfsSinkConnectorConfig;
@@ -28,8 +30,6 @@ import io.confluent.connect.storage.format.SchemaFileReader;
 import io.confluent.connect.storage.hive.HiveFactory;
 import org.apache.hadoop.fs.Path;
 
-import static io.confluent.connect.storage.StorageSinkConnectorConfig.SCHEMA_CACHE_SIZE_CONFIG;
-
 /**
  * Extended AvroFormat class to support custom AvroRecordWriter to allow writting key and value to
  * HDFS.
@@ -40,7 +40,8 @@ public class AvroFormatRadar implements Format<HdfsSinkConnectorConfig, Path> {
     public AvroFormatRadar(HdfsStorage storage) {
         AvroDataConfig avroConfig = new AvroDataConfig.Builder()
                 .with(AvroDataConfig.CONNECT_META_DATA_CONFIG, false)
-                .with(AvroDataConfig.SCHEMAS_CACHE_SIZE_CONFIG, storage.conf().getInt(SCHEMA_CACHE_SIZE_CONFIG))
+                .with(AvroDataConfig.SCHEMAS_CACHE_SIZE_CONFIG,
+                        storage.conf().getInt(SCHEMA_CACHE_SIZE_CONFIG))
                 .with(AvroDataConfig.ENHANCED_AVRO_SCHEMA_SUPPORT_CONFIG, true)
                 .build();
 

--- a/src/main/java/org/radarcns/sink/hdfs/AvroFormatRadar.java
+++ b/src/main/java/org/radarcns/sink/hdfs/AvroFormatRadar.java
@@ -17,6 +17,7 @@
 package org.radarcns.sink.hdfs;
 
 import io.confluent.connect.avro.AvroData;
+import io.confluent.connect.avro.AvroDataConfig;
 import io.confluent.connect.hdfs.HdfsSinkConnectorConfig;
 import io.confluent.connect.hdfs.avro.AvroFileReader;
 import io.confluent.connect.hdfs.avro.AvroHiveFactory;
@@ -37,7 +38,13 @@ public class AvroFormatRadar implements Format<HdfsSinkConnectorConfig, Path> {
     private final AvroData avroData;
 
     public AvroFormatRadar(HdfsStorage storage) {
-        this.avroData = new AvroData(storage.conf().getInt(SCHEMA_CACHE_SIZE_CONFIG));
+        AvroDataConfig avroConfig = new AvroDataConfig.Builder()
+                .with(AvroDataConfig.CONNECT_META_DATA_CONFIG, false)
+                .with(AvroDataConfig.SCHEMAS_CACHE_SIZE_CONFIG, storage.conf().getInt(SCHEMA_CACHE_SIZE_CONFIG))
+                .with(AvroDataConfig.ENHANCED_AVRO_SCHEMA_SUPPORT_CONFIG, true)
+                .build();
+
+        this.avroData = new AvroData(avroConfig);
     }
 
     public RecordWriterProvider<HdfsSinkConnectorConfig> getRecordWriterProvider() {

